src/lib/optimizer/strategy/predicate_reordering_rule.cpp

```diff
@@ -44,14 +44,11 @@ bool PredicateReorderingRule::apply_to(const std::shared_ptr<AbstractLQPNode>& n
     if (predicate_nodes.size() > 1) {
       reordered = _reorder_predicates(predicate_nodes);
       reordered |= _apply_to_inputs(predicate_nodes.back());
-    } else {
-      // No chain was found, continue with the current nodes inputren.
-      reordered = _apply_to_inputs(node);
     }
-  } else {
-    reordered = _apply_to_inputs(node);
   }
 
+  reordered = _apply_to_inputs(node);
```

Comment:
Shouldn't that be `_apply_to_inputs(node_at_the_end_of_the_predicate_chain_OR_the_input_node)`? The way it is done here, the rule gets applied multiple times to subsets of a predicate chain, right?